boards/rp-pico/examples/pico_display_pack_2.rs

```diff
@@ -0,0 +1,170 @@
+
+    // The single-cycle I/O block controls our GPIO pins
+    let sio = hal::Sio::new(pac.SIO);
+
+    // Set the pins up according to their function on this particular board
+    let pins = rp_pico::Pins::new(
+        pac.IO_BANK0,
+        pac.PADS_BANK0,
+        sio.gpio_bank0,
+        &mut pac.RESETS,
+    );
+
+    // Init PWMs
+    let mut pwm_slices = hal::pwm::Slices::new(pac.PWM, &mut pac.RESETS);
+
+    // Configure and enable PWMs
+    let pwm3 = &mut pwm_slices.pwm3;
+    pwm3.set_ph_correct();
+    pwm3.enable();
+    let pwm4 = &mut pwm_slices.pwm4;
+    pwm4.set_ph_correct();
+    pwm4.enable();
+
+    // Configure button inputs
+    // 12 - A button
+    let _a_pin = pins.gpio12.into_floating_input();
+    // 13 - B button
+    let _b_pin = pins.gpio13.into_floating_input();
+    // 14 - X button
+    let _x_pin = pins.gpio14.into_floating_input();
+    // 15 - Y button
+    let _y_pin = pins.gpio15.into_floating_input();
+
+    // 6 (PWM 3A) - LED red
+    let r_channel = &mut pwm3.channel_a;
+    r_channel.output_to(pins.gpio6);
+    r_channel.set_inverted();
+    r_channel.set_duty(0);
+    // 7 (PWM 3B) - LED green
+    let g_channel = &mut pwm3.channel_b;
+    g_channel.output_to(pins.gpio7);
+    g_channel.set_inverted();
+    g_channel.set_duty(0);
+    // 8 (PWM 4A) - LED blue
+    let b_channel = &mut pwm4.channel_a;
+    b_channel.output_to(pins.gpio8);
+    b_channel.set_inverted();
+    b_channel.set_duty(0);
+
+    // 20 - LCD backlight
+    pins.gpio20.into_push_pull_output().set_high().unwrap();
+
+    // 22 - LCD hard reset
+    let mut rst_pin = pins.gpio22.into_push_pull_output();
+    rst_pin.set_low().unwrap();
+
+    // Setup SPI
+    let spi_cs = pins.gpio17.into_push_pull_output();
+    let spi_miso = pins.gpio16.into_push_pull_output();
+    let _spi_sclk = pins.gpio18.into_mode::<hal::gpio::FunctionSpi>();
+    let _spi_mosi = pins.gpio19.into_mode::<hal::gpio::FunctionSpi>();
+
+    let spi = hal::Spi::<_, _, 8>::new(pac.SPI0);
+
+    // Exchange the uninitialised SPI driver for an initialised one
+    let spi = spi.init(
+        &mut pac.RESETS,
+        clocks.peripheral_clock.freq(),
+        16_000_000u32.Hz(),
+        &embedded_hal::spi::MODE_0,
+    );
+
+    // display interface abstraction from SPI and DC
+    let di = SPIInterface::new(spi, spi_miso, spi_cs);
+
+    // create driver
+    let mut display = ST7789::new(di, rst_pin, 320, 240);
+
+    // initialize
+    display.init(&mut delay).unwrap();
+
+    // set default orientation
+    display.set_orientation(Orientation::Landscape).unwrap();
+
+    // create a filled circle
+    let circle =
+        Circle::new(Point::new(100, 60), 120).into_styled(PrimitiveStyle::with_fill(Rgb565::RED));
+
+    // draw the circle on a black background
+    display.clear(Rgb565::BLACK).unwrap();
+    circle.draw(&mut display).unwrap();
+
+    loop {
+        continue; // keep optimizer from removing in --release
```

Comment:
I don't think this is still necessary. https://github.com/phil-opp/blog_os/issues/793